src/genie/libs/parser/iosxe/show_cts.py

```diff
@@ -1945,3 +1945,79 @@ def filter_lines(raw_output):
                 continue
 
         return cts_rb_permissions_dict
+
+
+# =========================
+# Schema for:
+#  * 'show cts ap sgt info {ap_name}'
+# =========================
+class ShowCtsApSgtInfoSchema(MetaParser):
+    """Schema for show cts ap sgt info {ap_name}."""
+
+    schema = {
+        "number_of_sgts_referred_by_the_ap": int,
+        Optional("sgts") : {
+            Optional(str) : {
+                Optional("policy_pushed_to_ap"): str,
+                Optional("no_of_clients"): int
+            }
+        }
+    }
+
+
+# =========================
+# Parser for:
+#  * 'show cts ap sgt info {ap_name}'
+# =========================
+class ShowCtsApSgtInfo(ShowCtsApSgtInfoSchema):
+    """Parser for show cts ap sgt info {ap_name}"""
+
+    cli_command = 'show cts ap sgt info {ap_name}'
+
+    def cli(self, ap_name="", output=None):
+        if output is None:
+            output = self.device.execute(self.cli_command.format(ap_name=ap_name))
+        else:
+            output = output
+
+        # Number of SGTs referred by the AP...............: 2
+        #
+        # SGT               PolicyPushedToAP       No.of Clients
+        # ------------------------------------------------------------
+        # UNKNOWN(0)        NO                     0
+        # DEFAULT(65535)    NO                     0
+
+        # Number of SGTs referred by the AP...............: 0
+        p_number_sgts = re.compile(r"^Number\s+of\s+SGTs\s+referred\s+by\s+the\s+AP...............:\s+(?P<value>\d+)$")
+
+        # SGT               PolicyPushedToAP       No.of Clients
+        p_sgt_header = re.compile(r"^SGT\s+PolicyPushedToAP\s+No.of\s+Clients$")
+
+        # ------------------------------------------------------------
+        p_sgt_delimiter = re.compile(r"^------------------------------------------------------------$")
+
+        # 10                NO                     1
+        p_sgt_row = re.compile(r"^(?P<sgt>\S+)\s+(?P<policy>\S+)\s+(?P<clients>\d+)$")
+
+        cts_ap_dict = {}
+
+        for line in output.splitlines():
+            line = line.strip()
+            if p_number_sgts.match(line):
+                match = p_number_sgts.match(line)
+                cts_ap_dict.update({ "number_of_sgts_referred_by_the_ap": int(match.group("value")) })
+                continue
+            elif p_sgt_header.match(line):
+                continue
+            elif p_sgt_delimiter.match(line):
+                continue
+            elif p_sgt_row.match(line):
+                match = p_sgt_row.match(line)
+                group = match.groupdict()
+                if not cts_ap_dict.get("sgts"):
+                    cts_ap_dict.update({ "sgts": {} })
```

Comment:
No big deal if not changed, however you can do this instead
``` python
sgts_dict = cts_ap_dict.setdefault("sgts", {})  # only sets the key/value if the key doesnt already exist
sgts_dict.update(
	{
		group["sgt"]: {
			"policy_pushed_to_ap": group["policy"].lower(), 
			"no_of_clients": int(group["clients"])
		}
	}
)

Reply:
I need to get in the habit of using `setdefault()` it is much cleaner. Thank you! Updated!